### ISSUE TITLE
React-UI-Starter Build Script breaks app

### ISSUE THREAD

**Issue by khaIilnafis** [solana-labs/wallet-adapter#269]
**Describe the bug**
Creating production builds using the react-ui-starter app with npm run build results in a failure to 'tree shake' the wallet-adapter dependencies

**To Reproduce**
1. Start from a fresh copy of the react-ui-starter
2. Install dependencies 
3. NPM start (from the hot reloaded app you can see things are functioning fine)
4. npm run build (all wallet-adapter dependencies fail to get bundled)

**Expected behavior**
The expected behavior is that the wallet-adapter suite of dependencies are correctly bundled with the rest of the application

**Screenshots**
![Screen Shot 2022-01-18 at 8 13 03 PM](https://user-images.githubusercontent.com/5186791/150045822-dec7c6e5-b74d-445b-a925-6595cd0684d8.png)
3S)
![Screen Shot 2022-01-18 at 8 20 45 PM](https://user-images.githubusercontent.com/5186791/150045840-d8455402-3d89-4a1c-8281-be5c6860452f.png)

**Desktop (please complete the following information):**
 - OS: MacOS Big Sur v11.6
 - Browser: Chrome
 - Version: 
 -      "@solana/wallet-adapter-base": "^0.9.2",
        "@solana/wallet-adapter-react": "^0.15.2",
        "@solana/wallet-adapter-react-ui": "^0.9.3",
        "@solana/wallet-adapter-wallets": "^0.14.2",

**Comment by dewittalex** [comment]
I have a similar problem. "yarn build" doesn't create a build folder but I still try to use the dist folder as a build (I'm very new and I don't know if this makes sense). When I deploy the dist, I have a similar output as Khalil where there is no styling and the buttons don't work.

But when I run yarn start, everything works as it should.

![Screenshot from 2022-01-18 20-29-14](https://user-images.githubusercontent.com/55413326/150046499-72238208-fe60-4b9b-bd2a-2d6081c74be4.png)

**Comment by jordansexton** [comment]
Interesting, thanks for the reports. Despite the claim @ https://github.com/parcel-bundler/parcel/issues/7165#issuecomment-950113952 it does appear that something is up with tree shaking -- for example, the Blocto SDK is being included in the build even when it's not used.

This sounds like it could be a bug in our config, but I'm not sure where to start with fixing it. Any ideas?

**Comment by khaIilnafis** [comment]
> Interesting, thanks for the reports. Despite the claim @ [parcel-bundler/parcel#7165 (comment)](https://github.com/parcel-bundler/parcel/issues/7165#issuecomment-950113952) it does appear that something is up with tree shaking -- for example, the Blocto SDK is being included in the build even when it's not used.
> 
> This sounds like it could be a bug in our config, but I'm not sure where to start with fixing it. Any ideas?

Reading through the parcel docs now. Will report back if I can get anything to work.

**Comment by steveluscher** [comment]
Interesting error! Something about this seems off though.

That import is definitely static. It imports `React` and `StrictMode`, every time, all the time.

A dynamic import would be something that's done conditionally.

```
if (doesLikeCheese) {
  const cheese = await import('velveeta');
  /* ... */
}
```

Or something where the import is programmatic.

```
const cheese = await import(`./cheeses/${cheeseType}`);
```

Neither of these apply here.

I'm inclined to think it's a bug in Parcel, as suggested by @mischnic in https://github.com/parcel-bundler/parcel/issues/7165#issuecomment-950113952. I'd be tempted to follow that issue and ignore it for now.

> …for example, the Blocto SDK is being included in the build even when it's not used.

I'm not so worried about blocto-sdk appearing as a bundle in your build output, because that _is_ something that we use as a dynamic import, here:

https://github.com/solana-labs/wallet-adapter/blob/master/packages/wallets/blocto/src/adapter.ts#L75-L79

Happy to be wrong about any of this and to learn something though!

**Comment by jordansexton** [comment]
> I'm not so worried about blocto-sdk appearing as a bundle in your build output

We never import the blocto adapter though. If tree shaking was working, we would never dynamically import its dependency.

https://github.com/solana-labs/wallet-adapter/blob/16fb5cc395e9974a91297549feaf7694191c7093/packages/starter/react-ui-starter/src/App.tsx#L4-L12

**Comment by steveluscher** [comment]
🧐 Oooh… interesting.

**Comment by steveluscher** [comment]
Something is definitely up, but I'm not too worried about it. I just checked out a fresh, fresh, fresh copy of this repo, and:

```
yarn install
yarn build
cd packages/starter/react-ui-starter
cd dist
python -m SimpleHTTPServer 8001 & open http://localhost:8001
```

If the Blocto wallet adapter _wasn't_ being tree-shaken, I would expect to be able to find the string `https://blocto.app` in the JavaScript somewhere, but it does not appear.

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/13243/150066677-43f12f42-c569-44a3-ae61-a2c4fe65b37e.png">

**Comment by khaIilnafis** [comment]
@steveluscher in the steps you've laid out you'll find that the WalletModal doesn't open when you click select & the module hasn't been loaded properly, i.e. all css is missing. So it still wasn't bundled correctly.

**Comment by dewittalex** [comment]
Does anyone know a quick fix or the version #s I should revert back to?

**Comment by steveluscher** [comment]
Got it! Thanks for the report. Fixed in #274.
